AK/Math.h

```diff
@@ -424,48 +432,70 @@ using Trigonometry::tan;
 namespace Exponentials {
 
 template<FloatingPoint T>
-constexpr T log(T x)
+constexpr T log2(T x)
 {
-    CONSTEXPR_STATE(log, x);
+    CONSTEXPR_STATE(log2, x);
 
 #if ARCH(X86_64)
     T ret;
     asm(
-        "fldln2\n"
+        "fld1\n"
         "fxch %%st(1)\n"
         "fyl2x\n"
         : "=t"(ret)
         : "0"(x));
     return ret;
 #else
 #    if defined(AK_OS_SERENITY)
-    // TODO: Add implementation for this function.
-    TODO();
+    if (x == 0)
```

Comment:
This implementation seems fine to me in line with the rest of the file, but I'll re-iterate the belief I've had for a long time: All of this implementation detail of the C standard math functions should be in a private header in LibC, and normal library code should be going through the standard interfaces  (or builtins) unless we have a *really* compelling reason to do so.

Reply:
To put it another way, AK/Math.h and friends have too many responsibilities. They provide:
- wrappers for builtins
- *implementations* of C standard math functions that cannot use the builtins themselves (because builtins are allowed to call into the C standard functions if they give up on constexpr evaulation).
- hand optimized assembly versions of common math functions that arguably should not be this easily accessible. Hand-optimized assembly functions should be used when you have a massive list of receipts, not handed out like candy to every application.